develop/migration_mt_django.rst

```diff
@@ -5,76 +5,185 @@
+In :ref:`mt_schema_migration` we discussed the framework-agnostic database changes required for using Citus in the multi-tenant use case. Here we investigate specifically how to migrate multi-tenant Django applications to a Citus storage backend with the help of the `django-multitenant <https://github.com/citusdata/django-multitenant>`_ library.
+
+
+
+This process will be in 5 steps:
+
+- Introducing the tenant column to models missing it that we want to distribute
+- Changing the primary keys of distributed tables to include the tenant column
+- Updating the models to use the :code:`TenantModelMixin`
+- Distributing the data
+- Updating the Django Application to scope queries
+
 
 Preparing to scale-out a multi-tenant application
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Initially you’ll start with all tenants placed on a single database node. Django's typical conventions make some assumptions about the data storage which limit scale-out options.
+Initially you’ll start with all tenants placed on a single database node. To be able to scale out django, some simple changes will have to be made to your models.
 
-In particular, the ORM introduces a pattern where you normalize data and split it into many distinct models each identified by a single ``id`` column (usually added implicitly by the ORM). For instance, consider this simplified model:
+Let's consider this simplified model:
 
 .. code-block:: python
 
   from django.utils import timezone
   from django.db import models
 
-  class Store(models.Model):
-    name = models.CharField(max_length=255)
-    url = models.URLField()
+  class Country(models.Model):
+      name = models.CharField(max_length=255)
+
+  class Account(models.Model):
+      name = models.CharField(max_length=255)
+      domain = models.CharField(max_length=255)
+      subdomain = models.CharField(max_length=255)
+      country = models.ForeignKey(Country, on_delete=models.CASCADE)
+
+
+  class Manager(models.Model):
+      name = models.CharField(max_length=255)
+      account = models.ForeignKey(Account, on_delete=models.CASCADE,
+                                  related_name='managers')
+
+
+  class Project(models.Model):
+      name = models.CharField(max_length=255)
+      account = models.ForeignKey(Account, related_name='projects',
+                                  on_delete=models.CASCADE)
+      managers = models.ManyToManyField(Manager)
+
+
+  class Task(models.Model):
+      name = models.CharField(max_length=255)
+      project = models.ForeignKey(Project, on_delete=models.CASCADE,
+                                  related_name='tasks')
+
+
+
+The tricky thing with this pattern is that in order to find all tasks for an account, you'll have to query for all of a account's project first. This becomes a problem once you start sharding data, and in particular when you run UPDATE or DELETE queries on nested models like task in this example.
```

Comment:
in order to find all tasks for an account, you'll have to query for all of a*n* account's project*s* first